### ISSUE TITLE
custom parameter in answer url for voice api

### ISSUE THREAD

**Issue by ahmedhashmi21** [Nexmo/nexmo-php#66]
Hi Team,
I am using Nexmo for sending sms and text-to-speech voice call.
everything is working fine but I need to add custom parameter in answer url json for voice api.

please check below code:
================================================
$call = $client->calls()->create([
    'to' => [[
        'type' => 'phone',
        'number' => '919619666397'
    ]],
    'from' => [
        'type' => 'phone',
        'number' => '919619666397'
    ],
    'answer_url' => ['https://www.efundbook.com/efb_nexmo/interest.json'],
]);
=================================================
In above code, there static json file in answer_url.
Any time I execute the code it says same speech on voice call.

I need to change speech dynamically.

can  I use php file for 'answer_url' that will echo custom json out put. so that my voice api will say hello with customer name.
basically i need to change json text dynamically .

Please help me to get this done, thanks in advance.

**Comment by mheap** [comment]
Hi @ahmedhashmi21,

You can use a PHP script as your `answer_url` to customise the response. So long as the URL called returns JSON, Nexmo does not mind how the response is generated.

Unfortunately, there is no way to replace `answer_url` with JSON, but that's a great suggestion that I'll put to our product team!

Hope that helps

**Comment by ahmedhashmi21** [comment]
Hi Michael,

Thanks for providing resolution on this. I have tried php script for answer_url but it didn't work for me.
Could you please provide php script example so that I can use the same.

Thanks

**Comment by ahmedhashmi21** [comment]
Hi Michael,

Thanks for providing code. but i have only last issue, when I execute code it send blank call it is reading file.
Please check my code.
$call = $client->calls()->create([
    'to' => [[
        'type' => 'phone',
        'number' => '919619666397'
    ]],
    'from' => [
        'type' => 'phone',
        'number' => '919619666397'
    ],
    'answer_url' => ['https://www.efundbook.com/efb_nexmo/dynamic_json.php'],
]);
-----------------------------
dynamic_json.php
<?php

echo json_encode([
  [
    'action' => 'talk', 
    'voiceName' => 'Russell', 
    'text' => 'Hi, This is Russel.'
  ]
]);
?>

If above code works then I am done with it. please revert on this I would really appreciate it.
Thanks

**Comment by mheap** [comment]
It looks as though you're using the same number for `to` and `from`. The `from` number must be a number that you own on Nexmo.

Your `answer_url` looks good to me though!

**Comment by ahmedhashmi21** [comment]
Okay, but same code is working when I replace answer_url with json file.

As you said, I have changed from number but still I am getting blank call.

**Comment by mheap** [comment]
Ah, sorry @ahmedhashmi21. Nexmo require a file with the content type `application/json`, and you're serving up `text/html`. Try this:

```php
<?php

header('Content-Type: application/json');

echo json_encode([
  [
    'action' => 'talk', 
    'voiceName' => 'Russell', 
    'text' => 'Hi, This is Russell'
  ]
]);```

**Comment by ahmedhashmi21** [comment]
Hi Michael,

Thank you very much, Its working for me. really appreciate it!!!

**Comment by mheap** [comment]
No problem @ahmedhashmi21, thanks for using Nexmo and please let us know if you have any other issues.

**Comment by BloomBitz** [comment]
hii can you tell me how to make  dynamic ncco in asp.net

**Comment by BloomBitz** [comment]
Code: To send Text to speech  
string NEXMO_CALL_UUID = Session["UUID"].ToString();
            string TEXT = "Hello I am Kimberly.This is a text to speech sample for voice alert message using nexmo";

            var result = Client.Call.BeginTalk(NEXMO_CALL_UUID, new Call.TalkCommand
            {
                text = TEXT,
                voice_name = "Kimberly"
            });

I have successfully done an outbound call but when i try to synthesized text to speech along with the call and  whenever  i execute this,result give me an exception "value can not be null.Parameter name is value.".
give me some suggestion
